mobile/components/DataEntry.tsx

```diff
@@ -47,72 +49,112 @@ const DataEntry = () => {
+      ) : (
+        <View style={styles.container}>
+          <Text style={[styles.headerText, { paddingLeft: 105, paddingRight: 105 }]}>New Entry</Text>
+          <Text style={styles.promptText}>Date:</Text>
+          <TextInput
+            value={date}
+            style={styles.fieldBox}
+            placeholder="MM/DD/YYYY"
+            autoFocus
+            keyboardType="numbers-and-punctuation"
+            onChangeText={date => setDate(date)}
+            editable={editable}
+          />
+          <Text style={styles.promptText}>Fog Net ID #:</Text>
+          <TextInput
+            style={styles.fieldBox}
+            placeholder="Fog Net ID"
+            onChangeText={fogNetID => setFogNetID(fogNetID)}
+            editable={editable}
+          />
+          <Text style={styles.promptText}>Cluster ID #:</Text>
+          <TextInput
+            style={styles.fieldBox}
+            placeholder="Cluster ID"
+            onChangeText={clusterID => setClusterID(clusterID)}
+            editable={editable}
+          />
+          <Text style={styles.promptText}>Model Name:</Text>
+          <TextInput
+            style={styles.fieldBox}
+            placeholder="Model Name"
+            onChangeText={fogNetModel => setFogNetModel(fogNetModel)}
+            editable={editable}
+          />
+          <Text style={styles.promptText}>Water Collected (L):</Text>
+          <TextInput
+            style={styles.fieldBox}
+            placeholder="0.0"
+            keyboardType="decimal-pad"
+            onChangeText={waterCollected => setWaterCollected(parseFloat(waterCollected))}
+            editable={editable}
+          />
+          <TouchableOpacity style={styles.submitButton}
+            onPress={async () => {
+              try {
+                if (!date.trim()) {
+                  alert('Please Enter Date');
+                  return;
+                } else if (dateIsValid(date) === false) {
+                  alert('Invalid Date' + "\n" + 'Make sure date is MM/DD/YYYY');
+                  return;
+                } else if (!fogNetID.trim()) {
+                  alert('Please Enter Fog Net ID');
+                  return;
+                } else if (!clusterID.trim()) {
+                  alert('Please Enter Cluster ID');
+                  return;
+                } else if (!fogNetModel.trim()) {
+                  alert('Please Enter Model Name');
+                  return;
+                } else if (isNaN(waterCollected)) {
+                  alert('Please Enter a Valid Amount of Water Collected');
+                  return;
+                } else if (dateIsValid(date) === true) {
+                  setShowConfirm(true);
+                  setEditable(false);
+                }
+              } catch (err) {
+                alert('Please complete all fields');
+              }
+            }}>
+            <Text style={styles.buttonText}>Submit</Text>
+          </TouchableOpacity>
+        </View>
+      )}
```

Comment:
I think this section could be consolidated a bit. Rather than one huge conditional and completely separate components that could result in a weird transition when the change occurs on slower devices, it might be better to conditionally style each component (ie the TextInputs to be not editable and appear like text components) and maintain the general structure of the pre-existing component